### PR TITLE
feat(FX-4591): Add ability to create new list from modal

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feat**

This PR solves [FX-4591]

Review app: [create-new-list-modal](https://create-new-list-modal.artsy.net/)

### Demo
https://user-images.githubusercontent.com/3513494/223359863-72656d0e-a11b-45c7-b5a0-08cc4e6df3fa.mp4

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FX-4591]: https://artsyproduct.atlassian.net/browse/FX-4591?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/SelectListsForArtworkModal.tsx [hunk 1]
```diff
@@ -14,75 +14,56 @@ import { useUpdateCollectionsForArtwork } from "Apps/CollectorProfile/Routes/Sav
...
 }
 
 export const SelectListsForArtworkModal: React.FC<SelectListsForArtworkModalProps> = ({
   me,
   artwork,
-  onClose,
-  onSave,
 }) => {
   const { t } = useTranslation()
+  const { state, dispatch, onSave } = useManageArtworkForSavesContext()
```
Comment (damassi) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/SelectListsForArtworkModal.tsx:34]: ty for removing of all of the props passing! much less brittle

Comment (dimatretyak) [reply src/Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/SelectListsForArtworkModal.tsx]: you're welcome 😅

**Review by nickskalkin — DISMISSED** [review]

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/CreateNewListModal.tsx [hunk 2]
```diff
@@ -117,15 +132,27 @@ const CreateNewListModal: React.FC<CreateNewListModalProps> = ({
...
                 <Media greaterThanOrEqual="sm">
-                  <Flex justifyContent="flex-end">
+                  <Flex
+                    justifyContent={onBackClick ? "space-between" : "flex-end"}
+                  >
+                    {!!onBackClick && (
+                      <Button variant="secondaryBlack" onClick={onBackClick}>
+                        {t("collectorSaves.createNewListModal.back")}
+                      </Button>
+                    )}
```
Comment (anandaroop) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/CreateNewListModal.tsx:149]: Thanks for adding this button here ([Ryan confirmed](https://artsy.slack.com/archives/C9SATFLUU/p1678099239933699?thread_ts=1677803388.476209&cid=C9SATFLUU) that we should always display the back/cancel button in both desktop and mweb contexts)

Do we also need to modify the behavior of the mweb version below on L175?

**Review by anandaroop — APPROVED** [review]

File: src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/CreateNewListModal.tsx (hunk 2 above)
Comment (dimatretyak) [reply src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/CreateNewListModal.tsx]: Addressed here: https://github.com/artsy/force/pull/12021/commits
Thank you for clarifications!

**Pull request merged**